### ISSUE TITLE
Nil pointer derefence when sending block

### ISSUE THREAD

**Issue by alexsporn** [iotaledger/iota-core#556]
```
2023-11-24T12:18:37.1446341Z nodeD > [Main - 8d7e73ed] BlockDAG.MissingBlockAttached: BlockID(12.1-nodeB:12)
2023-11-24T12:18:37.1446576Z panic: runtime error: invalid memory address or nil pointer dereference
2023-11-24T12:18:37.1446773Z [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x11284d9]
2023-11-24T12:18:37.1446778Z 
2023-11-24T12:18:37.1446859Z goroutine 487550 [running]:
2023-11-24T12:18:37.1447065Z github.com/iotaledger/iota-core/pkg/model.(*Block).Data(...)
2023-11-24T12:18:37.1447511Z 	/actions-runner/_work/iota-core/iota-core/pkg/model/block.go:85
2023-11-24T12:18:37.1447830Z github.com/iotaledger/iota-core/pkg/network/protocols/core.(*Protocol).SendBlock(...)
2023-11-24T12:18:37.1448135Z 	/actions-runner/_work/iota-core/iota-core/pkg/network/protocols/core/protocol.go:53
2023-11-24T12:18:37.1448684Z github.com/iotaledger/iota-core/pkg/protocol.(*Protocol).runNetworkProtocol.func1({0x64, 0x3c, 0x25, 0xea, 0xab, 0xab, 0xd8, 0x7b, 0xf2, 0xb4, ...}, ...)
2023-11-24T12:18:37.1448997Z 	/actions-runner/_work/iota-core/iota-core/pkg/protocol/protocol_network.go:25 +0xf9
2023-11-24T12:18:37.1449220Z github.com/iotaledger/hive.go/runtime/event.(*Event2[...]).Trigger.func1.1()
2023-11-24T12:18:37.1449707Z 	/home/runner/go/pkg/mod/github.com/iotaledger/hive.go/runtime@v0.0.0-20231122112629-bdf1cc39fba7/event/events.go:134 +0xb0
2023-11-24T12:18:37.1449929Z github.com/iotaledger/hive.go/runtime/workerpool.(*Task).run(0xc021299b00)
2023-11-24T12:18:37.1450472Z 	/home/runner/go/pkg/mod/github.com/iotaledger/hive.go/runtime@v0.0.0-20231122112629-bdf1cc39fba7/workerpool/task.go:48 +0xb6
2023-11-24T12:18:37.1450761Z github.com/iotaledger/hive.go/runtime/workerpool.(*WorkerPool).workerReadLoop(0xc00c82f880)
2023-11-24T12:18:37.1451315Z 	/home/runner/go/pkg/mod/github.com/iotaledger/hive.go/runtime@v0.0.0-20231122112629-bdf1cc39fba7/workerpool/workerpool.go:190 +0x3b
2023-11-24T12:18:37.1451589Z github.com/iotaledger/hive.go/runtime/workerpool.(*WorkerPool).worker(0xc00c82f880)
2023-11-24T12:18:37.1452164Z 	/home/runner/go/pkg/mod/github.com/iotaledger/hive.go/runtime@v0.0.0-20231122112629-bdf1cc39fba7/workerpool/workerpool.go:170 +0x69
2023-11-24T12:18:37.1452495Z created by github.com/iotaledger/hive.go/runtime/workerpool.(*WorkerPool).startWorkers in goroutine 487505
2023-11-24T12:18:37.1453022Z 	/home/runner/go/pkg/mod/github.com/iotaledger/hive.go/runtime@v0.0.0-20231122112629-bdf1cc39fba7/workerpool/workerpool.go:162 +0x32
2023-11-24T12:18:37.1453205Z FAIL	github.com/iotaledger/iota-core/pkg/tests	238.715s
2023-11-24T12:18:37.1453268Z FAIL
```

**Comment by karimodm** [comment]
Similar to #451

**Comment by jonastheis** [comment]
Most likely fixed with [48b1db2](https://github.com/iotaledger/iota-core/commit/48b1db2eb87d5717fe5cec53a0527046c7babee9) in #293
